litho-processor/src/main/java/com/facebook/litho/specmodels/model/DefaultLayoutSpecGenerator.java

```diff
@@ -47,9 +47,12 @@ public TypeSpec generate(LayoutSpecModel layoutSpecModel, EnumSet<RunMode> runMo
     final TypeSpec.Builder typeSpec =
         TypeSpec.classBuilder(layoutSpecModel.getComponentName())
             .superclass(ClassNames.COMPONENT)
-            .addOriginatingElement((TypeElement) layoutSpecModel.getRepresentedObject())
             .addTypeVariables(layoutSpecModel.getTypeVariables());
 
+    if (SpecModelUtils.isTypeElement(layoutSpecModel)) {
```

Comment:
@passy @IanChilds Not sure if this is something that adheres to Litho's code style. If not I could extract that to a `boolean` and use that in the `if` condition. 
I also thought about creating a `TypeSpecHolder` generator responsible to add the `OriginatingElement`, but that would seem "overkill" in that case. If that's a better approach in your opinion though, let me know and I good get something up.

Reply:
It looks good to me. Thanks putting so much thought in it. :)